src/Entity/DatasetSubmission.php

```diff
@@ -2764,14 +2754,38 @@ public function getDistributionPoints()
         return $this->distributionPoints;
     }
 
+     /**
+     * Getter for the (first) erddap url link.
+     *
+     * @return DatasetLink|null
+     */
+    private function getErdappUrlLink() : ?DatasetLink
+    {
+        $datasetLinks = $this->getDatasetLinks()->filter(function (DatasetLink $datasetLink) {
+            return $datasetLink->getName() == "ERDDAP";
```

Comment:
Can use the constant LINK_NAME_CODES from DatasetLinks Entity

Reply:
Good one!